src/store/user/provider.tsx

```diff
@@ -1,8 +1,8 @@
 import React, { useEffect, useState } from "react";
-import { setAuthToken } from "../../functions/setAuthToken";
-import { singin } from "../../functions/singin";
+
+import { singinViaToken } from "services/signinViaToken";
+import { User } from "types/user/interfaces";
 import { UserContext } from "./context";
-import { User } from "./interface";
 
 export const LoginProvider: React.FunctionComponent<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
I would prefer something like that:
`type LoginProviderProps = PropsWithChildren<>;`
`export const LoginProvider = ({children}: LoginProviderProps):JSX.ELEMENT => {...`
You don't need to strictly say what you're returning. But if u wanna - okay.
But you should say what are you retuning in methods.